_posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc

```diff
@@ -0,0 +1,2334 @@
+working code.
+
+How to run the code locally will be very different depending on the type of code you are working with:
+
+Application Code::
+  You can typically spin up a local environment for application code to test it out. For example, if you had a simple
+  web server written in a general purpose programming language such as Ruby, you can run the server code to bring up a
+  local copy of the application that you can interact with (e.g `ruby web-server.rb`). You can then manually test it by
+  loading the web server in the browser. Alternatively, you could run the automated test suite associated with your
+  application (e.g `ruby web-server-test.rb`). The point is that (almost) everything can be done locally for fast
+  iteration.
+
+Infrastructure Modules::
+  You will need to bring up live infrastructure to test infrastructure code. Unlike with application code, there is no
+  way to have a true and complete local copy of a cloud. Therefore, the only way to know for sure your infrastructure
+  code works is by making the actual API calls to the cloud to deploy it. With infrastructure modules, this involves
+  deploying the module into a sandbox environment. For example, to test a terraform module, you can define example code
+  that sets up the necessary resource dependencies that the module needs, and then deploy that into your sandbox with
+  `terraform apply`. You can then inspect the deployed resources to make sure they are functioning as expected. For
+  convenience, this process could be captured in an automated test using a framework such as
+  https://terratest.gruntwork.io/[Terratest].
+
+Live Infrastructure Config::
+  Locally testing live infrastructure config is more difficult than either application code or infrastructure modules.
+  Unlike with infrastructure modules, it is difficult to deploy the live infrastructure config temporarily as the code
+  is tied to a specific live environment by nature of the code. After all, this is the configuration to manage live
+  infrastructure. To illustrate this point, consider a scenario where you are working on updating the cross account IAM
+  roles to access your environments, and you are at the point of reflecting your changes to prod. Would you want to
+  deploy that code to your live production environment off of an unreviewed branch? +
+  The only real test you can do for live infrastructure config is to do a dry run of your infrastructure code. Most Infrastructure as Code tools support a dry run of the code to check what it would do against your environment. For example, with Terraform, you could run `terraform plan` to sanity check the planned actions Terraform will take. This is especially useful for sanity checking a fresh clone of the code. The trunk should be a true reflection of the live environment, so you should expect there to be no changes to make on a fresh clone of trunk.
```

Comment:
I tried a few ways, but in the end could not figure out a way to do this in the definition list. I think I need to switch from definition lists, but then it makes it hard to do the switch over from side-by-side-comp => overall explanation at the end.

I think I am going to keep as is for now, and then revisit later, since I am already overbudget on this project and guide.